.github/CONTRIBUTING.md

```diff
@@ -1,17 +1,67 @@
- 1. Make sure your PR follows our [code guidelines](https://github.com/triplea-game/triplea/blob/master/docs/dev/code_standards.md)
- 2. Make sure your PR follows our [code formatting guidelines](https://github.com/triplea-game/triplea/blob/master/docs/dev/code_format.md)
- 3. Make sure you understand our [process of reviewing](https://github.com/triplea-game/triplea/blob/master/docs/dev/code_reviews.md)
-
-Every PR is validated with travis, an automated build service.
-If this build fails (and it's not because of some occasional server outages),  you are required to change something so the build passes.
-
-## Issues
-Open an issue if you want to report a bug or request a feature.
-If it's a severe bug we are probably going to fix it pretty fast but feature requests have a pretty low priority in comparison.
-Nobody is getting paid for this, so please be aware that we are not always able to respond on the same day as we are all doing this in our free time.
+Contributions are really welcome. See the [dev-setup](/docs/development/README.md)
+for more information on how to get started, guidelines, conventions, and 
+[areas of need](/docs/development/areas-of-need.md)
+
+
+
+# Contribution Guidelines
+
+- TripleA is deceptively difficult to work with:
+  - quite brittle, easy to break things
+  - tangled, lots needs to be done and difficult to do so in bite-sized iterations
+  - design is sometimes hasty and primitive
+  - non-obvious dependencies, reflection and serialization compatibility can
+  break things. For example re-ordering or renaming variables and methods
+  can break compatibility between TripleA Versions.
+
+## Testing Responsibility
+Each contributor should test their changes thoroughly. We wish to minimize
+problems when it comes time for releases. The goal is small, incremental,
+and solid changes.
+
+TripleA lacks automated testing and the manual testing overhead is large.
+For now the manual testing needs to be done to ensure we keep a solid
+codebase going forward.
+
+Automated tests are hugely welcome, TripleA needs lots more so that the
+manual testing can be reduced and changes made with confidence.
+
+## Things to watch out for - Compatibility Breakers
+There are three ares to be concerned about compatibility:
+1. Save game serialization. Elements of `GameData.java` are serialized to a file
+and then loaded again. Private variable changes here can break things. To test this,
+save a game file, make changes, then verify that the save game can be loaded.
+1. Network: a custom variant of RMI (remote method invocation) is used. To test
+this, start a game without any changes, host. Then in another instance, make
+changes, launch, and connect to the host and play the game through to a combat round.
+That should test most functionality to verify network compatibility
+1. Map XML loading: most of this is now explicit and is less brittle. After making updates,
```

Comment:
Codacy found an issue: [[ordered-list-marker-value] Marker should be `3`, was `1`](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6206681)